assets/js/embed.js

```diff
@@ -21,85 +14,49 @@ function get_playlist(plid, retries) {
-                            url.searchParams.set('autoplay', '1');
-                        }
-
-                        if (video_data.params.listen !== video_data.preferences.listen) {
-                            url.searchParams.set('listen', video_data.params.listen);
-                        }
-
-                        if (video_data.params.speed !== video_data.preferences.speed) {
-                            url.searchParams.set('speed', video_data.params.speed);
-                        }
-
-                        if (video_data.params.local !== video_data.preferences.local) {
-                            url.searchParams.set('local', video_data.params.local);
-                        }
-
-                        location.assign(url.pathname + url.search);
-                    });
-                }
-            }
+    helpers.xhr('GET', plid_url, {retries: 5, entity_name: 'playlist'}, {
+        on200: function (response) {
+            if (!response.nextVideo)
+                return;
+
+            player.on('ended', function () {
+                var url = new URL('https://example.com/embed/' + response.nextVideo);
+
+                url.searchParams.set('list', plid);
+                if (!plid.startsWith('RD'))
+                    url.searchParams.set('index', response.index);
+                if (video_data.params.autoplay || video_data.params.continue_autoplay)
+                    url.searchParams.set('autoplay', '1');
+                if (video_data.params.listen !== video_data.preferences.listen)
+                    url.searchParams.set('listen', video_data.params.listen);
+                if (video_data.params.speed !== video_data.preferences.speed)
+                    url.searchParams.set('speed', video_data.params.speed);
+                if (video_data.params.local !== video_data.preferences.local)
+                    url.searchParams.set('local', video_data.params.local);
+
+                location.assign(url.pathname + url.search);
+            });
         }
-    };
-
-    xhr.onerror = function () {
-        console.warn('Pulling playlist failed... ' + retries + '/5');
-        setTimeout(function () { get_playlist(plid, retries - 1); }, 1000);
-    };
-
-    xhr.ontimeout = function () {
-        console.warn('Pulling playlist failed... ' + retries + '/5');
-        get_playlist(plid, retries - 1);
-    };
-
-    xhr.send();
+    });
 }
 
-window.addEventListener('load', function (e) {
+addEventListener('load', function (e) {
     if (video_data.plid) {
         get_playlist(video_data.plid);
     } else if (video_data.video_series) {
         player.on('ended', function () {
             var url = new URL('https://example.com/embed/' + video_data.video_series.shift());
 
-            if (video_data.params.autoplay || video_data.params.continue_autoplay) {
+            if (video_data.params.autoplay || video_data.params.continue_autoplay)
```

Comment:
I'm in favor of keeping the curly brackets.

What do you think @SamantazFox?

Reply:
I have no preference, tbh. If you want to keep them, let's keep them.

Reply:
Well I would prefer to keep them.
@AHOHNMYC if thats possible

Reply:
@unixfox Of course this is possible, but I'd better remove them in this case, because with `{}`s this series of homogenous commands will grow up by 33% with no profit.
I'd prefer keeping brackets in more ambiguous cases (like two cascaded if's described below), but this one is simple and straight.

Reply:
Wouldn't it be better to think about minifying before compiling invidious? This way we can keep a clean code base but in the final js files everything would be minified in order to take as less size as possible.

Reply:
@unixfox It is not about minifying, I just don't see any profit of {} there :) But if you want, i'll restore them.

Also IMO minificaton is evil. All scripts are gzipped, cached on browser side and weight less than 5 seconds of any video. Minification justified on webapps like Element (former Riot). But it is not voice for minifier. It is scream about dirty code and tools in JS world.

The only thing that will changed with minification — end users will have less transparency about executed code and less ability to modify it with userscripts in *monkey browser extensions.

Reply:
If the user wants to see the original source code, we can just provide it through source maps: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/SourceMap